### PR TITLE
Add support for Date type to Kudu connector

### PR DESCRIPTION
**Pull request by pgasp**
## Description
Adding Date Support to kudu connector

Relates to #11009

## Release notes

(x) Release notes are required, with the following suggested text:

```markdown
# Kudu
* Map Kudu `date` type to Trino `date` type. Previously, it was mapped to `varchar` type. ({issue}`18999`)
```

### CODE REVIEW

File: plugin/trino-kudu/src/main/java/io/trino/plugin/kudu/KuduPageSink.java [hunk 1]
```diff
@@ -153,6 +154,9 @@ private void appendColumn(PartialRow row, Page page, int position, int channel,
         else if (TIMESTAMP_MILLIS.equals(type)) {
             row.addLong(destChannel, truncateEpochMicrosToMillis(TIMESTAMP_MILLIS.getLong(block, position)));
         }
+        else if (DATE.equals(type)) {
+            row.addDate(destChannel, DateUtil.epochDaysToSqlDate(INTEGER.getInt(block, position)));
```
Comment (ebyhr) [inline plugin/trino-kudu/src/main/java/io/trino/plugin/kudu/KuduPageSink.java:158]: Static import `epochDaysToSqlDate`.

File: plugin/trino-kudu/src/main/java/io/trino/plugin/kudu/TypeHelper.java [hunk 2]
```diff
@@ -120,11 +120,13 @@ private static Type fromKuduClientType(org.apache.kudu.Type ktype, ColumnTypeAtt
                 return BooleanType.BOOLEAN;
             case BINARY:
                 return VarbinaryType.VARBINARY;
+            case DATE:
+                return DateType.DATE;
```
Comment (ebyhr) [inline plugin/trino-kudu/src/main/java/io/trino/plugin/kudu/TypeHelper.java:124]: Move under `DECIMAL`.

File: plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduIntegrationDateColumns.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
+/*
```
Comment (ebyhr) [inline plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduIntegrationDateColumns.java:1]: No need to add a new class. Please update the existing test class. e.g. BaseKuduConnectorSmokeTest

File: plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduIntegrationDateColumns.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        if (kuduServer != null) {
+            kuduServer.close();
+            kuduServer = null;
+        }
+    }
+
+    @Test
+    public void testCreateTableWithIntegerColumn()
+    {
+        for (TestDate test : TEST_DATE) {
```
Comment (ebyhr) [inline plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduIntegrationDateColumns.java:55]: Why do we need to use `for` look for one entry?

File: plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduIntegrationDateColumns.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+            doTestCreateTableWithIntegerColumn(test);
+        }
+    }
+
+    private void doTestCreateTableWithIntegerColumn(TestDate test)
```
Comment (ebyhr) [inline plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduIntegrationDateColumns.java:60]: Inline.

File: plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduIntegrationDateColumns.java [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+    {
+        String dropTable = "DROP TABLE IF EXISTS test_date";
+        String createTable = "" +
```
Comment (ebyhr) [inline plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduIntegrationDateColumns.java:63]: Please cover `CREATE TABLE AS SELECT` statement.

File: plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduIntegrationDateColumns.java [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+                "  datecol " + test.type + "\n" +
+                ") WITH (\n" +
+                " partition_by_hash_columns = ARRAY['id'],\n" +
+                " partition_by_hash_buckets = 2\n" +
+                ")";
+
+        assertUpdate(dropTable);
+        assertUpdate(createTable);
+
+        assertUpdate("INSERT INTO test_date VALUES(1, DATE '1977-07-14')", 1);
```
Comment (ebyhr) [inline plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduIntegrationDateColumns.java:75]: Please add more data. e.g. min & max date in Kudu, DST, julian->gregorian switch, null.

File: plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduIntegrationDateColumns.java [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+        MaterializedResult result = computeActual("SELECT id, datecol FROM test_date");
+        assertEquals(result.getRowCount(), 1);
+        Object obj = result.getMaterializedRows().get(0).getField(1);
+        assertTrue(obj instanceof LocalDate);
```
Comment (ebyhr) [inline plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduIntegrationDateColumns.java:79]: Please use `assertQuery` or `assertQuery(query(...)).matches()` method.

**Review by ebyhr — COMMENTED** [review]
Just skimmed.

**Comment by ebyhr** [comment]
https://github.com/trinodb/trino/pull/8229 is the previous attempt.

File: plugin/trino-kudu/src/main/java/io/trino/plugin/kudu/KuduPageSink.java (hunk 1 above)
Comment (pgasp) [reply plugin/trino-kudu/src/main/java/io/trino/plugin/kudu/KuduPageSink.java:158]: done

File: plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduIntegrationDateColumns.java (hunk 3 above)
Comment (pgasp) [reply plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduIntegrationDateColumns.java:1]: The existing class are handling specific type such as Integer or Decimal. To keep code clear I build dedicated class on Date only.

Comment (ebyhr) [reply plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduIntegrationDateColumns.java:1]: This is different from how we manage test code. Please use the existing test class.

File: plugin/trino-kudu/src/main/java/io/trino/plugin/kudu/TypeHelper.java [hunk 9]
```diff
@@ -122,9 +122,10 @@ private static Type fromKuduClientType(org.apache.kudu.Type ktype, ColumnTypeAtt
                 return VarbinaryType.VARBINARY;
             case DECIMAL:
                 return DecimalType.createDecimalType(attributes.getPrecision(), attributes.getScale());
-            // TODO: add support for varchar and date types: https://github.com/trinodb/trino/issues/11009
-            case VARCHAR:
+            // TODO: add support for varchar types: https://github.com/trinodb/trino/issues/11009
```
Comment (ebyhr) [inline plugin/trino-kudu/src/main/java/io/trino/plugin/kudu/TypeHelper.java:125]: The comment place is wrong now. Please move to above `case VARCHAR:`

**Review by ebyhr — COMMENTED** [review]
Please rebase on master instead of "Merge" and also squash commits into one.

File: plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduIntegrationDateColumns.java (hunk 7 above)
Comment (pgasp) [reply plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduIntegrationDateColumns.java]: Null value in Kudu Connector is not working for any Type. I will fix the problem in an other PR

File: plugin/trino-kudu/src/main/java/io/trino/plugin/kudu/TypeHelper.java (hunk 9 above)
Comment (pgasp) [reply plugin/trino-kudu/src/main/java/io/trino/plugin/kudu/TypeHelper.java:125]: Done

**Comment by ebyhr** [comment]
Superseded by https://github.com/trinodb/trino/pull/20086

**Pull request closed without merging**